src/app.ts

```diff
@@ -166,7 +165,31 @@ export class App extends Construct {
       default:
         break;
     }
+
   }
+
+  /**
+   * Synthesizes the app into a YAML string.
+   *
+   * @returns A string with all YAML objects across all charts in this app.
+   */
+  public synthYaml(): any {
+    validate(this);
+
+    var yamls: string[] = [];
+    const charts = this.charts;
+
+    for (const chart of charts) {
+      const apiObjects = chartToKube(chart);
+
+      apiObjects.forEach((apiObject) => {
+        yamls.push(Yaml.stringify(apiObject.toJson()));
+      });
+    }
+
+    return yamls.join('---\n');
```

Comment:
This bit of logic (stringifying objects and then joining them with "---\n") seems to overlap with the the logic in `Yaml.save()` - can we please extract these parts out to a common function (e.g. `formatObjects()`) that can be used in both places?

Reply:
Added this function. I will still have to return it with `.join('---\n')` since the function does not add it to the end of every object.